pkg/jx/cmd/install.go

```diff
@@ -381,151 +383,229 @@ func (options *InstallOptions) Run() error {
+		return errors.Wrap(err, "getting the platform version")
+	}
+
+	if options.Flags.GitOpsMode {
+		err := options.installPlatformGitOpsMode(gitOpsEnvDir, gitOpsDir, configStore, DEFAULT_CHARTMUSEUM_URL,
+			JenkinsXPlatformChartName, ns, version, valuesFiles, secretsFiles)
+		if err != nil {
+			return errors.Wrap(err, "installing the Jenkins X platform in GitOps mode")
 		}
-		err := options.installMissingDependencies(deps)
+	} else {
+		err := options.installPlatform(providerEnvDir, JenkinsXPlatformChart, JenkinsXPlatformRelease,
+			ns, version, valuesFiles, secretsFiles)
 		if err != nil {
-			log.Errorf("%v\nPlease fix the error or install manually then try again", err)
-			os.Exit(-1)
+			return errors.Wrap(err, "installing the Jenkins X platform")
 		}
 	}
 
-	if options.Flags.Provider == AKS {
-		var deps []string
-		d := binaryShouldBeInstalled("az")
-		if d != "" {
-			deps = append(deps, d)
-		}
-		err := options.installMissingDependencies(deps)
+	if options.Flags.CleanupTempFiles {
+		err := options.cleanupTempFiles(temporaryFiles)
 		if err != nil {
-			log.Errorf("%v\nPlease fix the error or install manually then try again", err)
-			os.Exit(-1)
+			return errors.Wrap(err, "cleaning up the temporary files")
 		}
 	}
 
-	initOpts := &options.InitOptions
-	helmBinary := initOpts.HelmBinary()
+	err = options.configureProwInTeamSettings()
+	if err != nil {
+		return errors.Wrap(err, "configuring Prow in team settings")
+	}
 
-	// configure the Helm binary
-	options.Helm().SetHelmBinary(helmBinary)
-	if initOpts.Flags.NoTiller {
-		helmer := options.Helm()
-		helmCli, ok := helmer.(*helm.HelmCLI)
-		if ok && helmCli != nil {
-			options.helm = helm.NewHelmTemplate(helmCli, helmCli.CWD, client, originalNs)
-		} else {
-			helmTemplate, ok := helmer.(*helm.HelmTemplate)
-			if ok {
-				options.helm = helmTemplate
-			} else {
-				log.Warnf("Helm facade is not a *helm.HelmCLI or *helm.HelmTemplate: %#v\n", helmer)
-			}
-		}
+	err = options.configureTillerInDevEnvironment()
+	if err != nil {
+		return errors.Wrap(err, "configuring Tiller in the dev environment")
 	}
 
-	dependencies := []string{}
-	if !initOpts.Flags.RemoteTiller && !initOpts.Flags.NoTiller {
-		binDir, err := util.JXBinLocation()
-		if err != nil {
-			return errors.Wrap(err, "reading jx bin location")
-		}
-		_, install, err := shouldInstallBinary("tiller")
-		if !install && err == nil {
-			confirm := &survey.Confirm{
-				Message: "Uninstalling existing tiller binary:",
-				Default: true,
-			}
-			flag := true
-			err = survey.AskOne(confirm, &flag, nil)
-			if err != nil || flag == false {
-				return errors.New("Existing tiller must be uninstalled first in order to use the jx in tiller less mode")
-			}
-			// Uninstall helm and tiller first to avoid using some older version
-			err = options.UninstallBinary(binDir, "tiller")
-			if err != nil {
-				return errors.Wrap(err, "uninstalling existing tiller binary")
-			}
-		}
+	err = options.configureHelm3(ns)
```

Comment:
helm3? How does this differ from Helm? Comment needed, or rename the method to `configureHelm` if we _only_ install helm?

Reply:
I think, we configure helm 3 before running the addon. I don't what to do a deeper refactoring here. @jstrachan Do we still need helm3 for addons?

Reply:
we still need helm3 - docs on [no tiller](https://jenkins-x.io/news/helm-without-tiller/) and [helm3](https://jenkins-x.io/architecture/helm3/)

when helm3 is sufficiently GA so most users are happy to switch we'll be able to make it the default and remove the tiller/helm2 stuff but thats a way off